org.sil.scripture-app-builder.yml

```diff
@@ -0,0 +1,114 @@
+# REFERENCES:
+  # Python2.7:  https://github.com/flathub/shared-modules/blob/cc50bdbb34cc3d29bf61cfea5e27cd980b7bda04/python2.7/python-2.7.json
+  # OpenJDK17: https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk17/blob/branch/23.08/org.freedesktop.Sdk.Extension.openjdk17.yaml
+  # Node20:    https://github.com/flathub/org.freedesktop.Sdk.Extension.node20/blob/branch/23.08/org.freedesktop.Sdk.Extension.node20.yaml
+# FIRST-TIME-BUILD: ~30m on 10-20Mb/s download speed
+app-id: org.sil.scripture-app-builder
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+  # - org.freedesktop.Sdk.Extension.openjdk21
+finish-args:
+  # Window x11 socket access
+  - --socket=x11
+  # Audio access
+  - --socket=pulseaudio
+  # Unsure if the program absolutely needs ssh
+  # - --socket=ssh-auth
+  # Or the printer socket
+  # - --socket=cups
+  # Network access
+  - --share=network
+  # XShm access
+  - --share=ipc
+  # All external devices (catch all for devices we interact with and graphical acceleration)
+  - --device=all
+  # Local file browsing access
+  - --filesystem=host
+  # Temporary file access
+  - --filesystem=/tmp
+  # Filesystem mounts
+  - --filesystem=/mnt
+  # USB Drives
+  - --system-talk-name=org.freedesktop.Udisks2
+  # File saving access
+  - --system-talk-name=org.freedesktop.FileManager1
+
+# Build args give resource access during the build
+build-options:
+  build-args:
+    - --share=network
+
+# App's run command
+command: sab-run.sh
+
+modules:
+
+# For in-app terminal
+- xterm.yml
+
+- python3-aeneas.yml
+
+# For python scripts
+- python3-requests.yaml
+
+# For pwa
+- sab-node20.yml
+
+- openjdk17.yml
+
+- name: scripture-app-builder
+  buildsystem: simple
+  build-commands:
+    - pwd
+    # Source tarball from a docker image and extract to flatpak app
+    - cd /tmp
+    - python3 /run/build/scripture-app-builder/docker_pull.py ghcr.io/sillsdev/scripture-app-builder:12.1
+    - python3 /run/build/scripture-app-builder/extract_layer.py /app/output
```

Comment:
why pull from a docker image? Where is the source of the application ?

Reply:
Fwiw this will not work on Flathub, there is no network access during build.

Reply:
Due to relationships with Bible societies, the source code is in a private repo.  We do a Build in GitHub Actions which creates a tarball of the artifacts and imports that into a docker image. Since we have a nonprofit GitHub account, there is no cost for storing this in GHCR.  It is an economical way to store the results of a build to share and is versioned with tags.

[Saving Images and Containers as Tar Files for Sharing](https://dockerlabs.collabnix.com/beginners/saving-images-as-tar/)

Here is the step we use in our GitHub Action:
```
      - name: "Push AppBuilders Tarball to GHCR"
        run: |
          docker import release/output/app-builders.tgz app-builders:latest
          docker tag app-builders:latest ghcr.io/$OWNER/app-builders:latest
          docker tag app-builders:latest ghcr.io/$OWNER/app-builders:${{ steps.version.outputs.SAB }}
          docker push ghcr.io/$OWNER/app-builders:latest
          docker push ghcr.io/$OWNER/app-builders:${{ steps.version.outputs.SAB }}
```

I don't quite understand `there is no network access during build`. How would installing python modules or OpenJFX into the Flatpak image work without network access?

Reply:
So we can't specify a docker image to download and then extract the files?  It needs to be a zip file hosted somewhere with a sha256?

Reply:
docker images are not an archive format.

Reply:
Will change to tar ball stored in an S3 bucket.

Reply:
Removed and using zip from S3 bucket.